### PR TITLE
[Web] User FS (user://) now correctly uses project name.

### PR DESCRIPTION
**Pull request by Faless**
This allows multiple instances to co-exist in the same domain while keeping their user data separate (each in its own folder).

### CODE REVIEW

File: platform/web/os_web.cpp [hunk 1]
```diff
@@ -157,7 +158,22 @@ void OS_Web::vibrate_handheld(int p_duration_ms) {
...
+			if (custom_dir.is_empty()) {
+				custom_dir = appname;
+			}
+			return userfs.path_join(custom_dir).replace("\\", "/");
+		} else {
+			return userfs.path_join(get_godot_dir_name()).path_join("app_userdata").path_join(appname).replace("\\", "/");
+		}
+	}
+
+	return userfs.path_join(get_godot_dir_name()).path_join("app_userdata").path_join("[unnamed project]");
```
Comment (akien-mga) [inline platform/web/os_web.cpp:176]: Maybe we should move this logic in `OS::get_user_data_dir()` and remove it from Linux, Windows, and here? And just set `get_data_path()` to `/userfs` here. (The only difference between Linux and Windows implementations seems to be the `replace("\\", "/")` from Windows, which you're also using here. They probably don't hurt to have applied for Linux too?)

Unless `get_data_path()` should say `/home/web_user/.local/share` here and thus we can't rely on a common implementation using `get_data_path()`. In that case, maybe the whole appname/custom dir logic could be moved to a helper method in `ProjectSettings`?

**Review by akien-mga — APPROVED** [review]
Suggested some refactoring but it's not blocking, this looks good as is.

File: platform/web/os_web.cpp (hunk 1 above)
Comment (Faless) [reply platform/web/os_web.cpp:176]: Yeah, I feel like at least the appname/custom dir logic should be moved, I'll look into it.

**Comment by akien-mga** [comment]
Thanks!

**Comment by dsnopek** [comment]
:tada: I am so happy this managed to squeezed in before 4.0!

**Pull request merged**